L1Trigger/L1TTwinMux/src/DTRPCBxCorrection.cc

```diff
@@ -0,0 +1,211 @@
+//-------------------------------------------------
+//
+//   Class: DTRPCBxCorrection
+//
+//   DTRPCBxCorrection
+//
+//
+//   Author :
+//   G. Flouris               U Ioannina    Mar. 2015
+//modifications:  G Karathanasis   U Athens
+//--------------------------------------------------
+#include <iostream>
+#include <iomanip>
+#include <iterator>
+
+#include "L1Trigger/L1TTwinMux/interface/DTRPCBxCorrection.h"
+#include "L1Trigger/L1TTwinMux/interface/L1MuTMChambPhContainer.h"
+
+using namespace std;
+
+DTRPCBxCorrection::DTRPCBxCorrection(L1MuDTChambPhContainer inphiDTDigis, L1MuDTChambPhContainer inphiRPCDigis) :m_phiDTDigis(inphiDTDigis),m_phiRPCDigis(inphiRPCDigis) {
+//  m_phiDTDigis=inphiDTDigis;
+//  m_phiRPCDigis=inphiRPCDigis;
+};
+
+
+void DTRPCBxCorrection::run( const edm::EventSetup& c) {
+
+  const L1TTwinMuxParamsRcd& tmParamsRcd = c.get<L1TTwinMuxParamsRcd>();
+  tmParamsRcd.get(tmParamsHandle);
+  const L1TTwinMuxParams& tmParams = *tmParamsHandle.product();
+
+  m_QualityLimit = tmParams.get_USERPCBXFORDTBELOWQUALITY();
+  m_DphiWindow   = tmParams.get_DphiWindowBxShift();
+
+  BxCorrection(0);
+  BxCorrection(1);
+
+  m_dt_tsshifted.setContainer(m_l1ttma_out);
+}
+
+void DTRPCBxCorrection::BxCorrection(int track_seg){
+
+  L1MuDTChambPhDigi * dtts=nullptr;
+  L1MuDTChambPhDigi * dttsnew=nullptr;
+  L1MuDTChambPhDigi * rpcts1=nullptr;
+  
+  auto m_phiDTDigis_tm=std::make_shared<L1MuTMChambPhContainer>();
+  //std::shared_ptr<L1MuTMChambPhContainer> m_phiDTDigis_tm (new L1MuTMChambPhContainer);
+  const std::vector<L1MuDTChambPhDigi> *phiChambVectorDT;
+  phiChambVectorDT= m_phiDTDigis.getContainer();
+  m_phiDTDigis_tm->setContainer(*phiChambVectorDT);
+  auto m_phiRPCDigis_tm=std::make_shared<L1MuTMChambPhContainer>();
+  //std::shared_ptr<L1MuTMChambPhContainer> m_phiRPCDigis_tm (new L1MuTMChambPhContainer);
+  const std::vector<L1MuDTChambPhDigi> *phiChambVectorRPC;
+  phiChambVectorRPC= m_phiRPCDigis.getContainer();
+  m_phiRPCDigis_tm->setContainer(*phiChambVectorRPC);
+
+  int bx=0, wheel=0, sector=0, station=1;
+  int ibx_dtm = 0, fbx_dtm = 0;
+  int ibx_dtp = 0, fbx_dtp = 0;
+
+  for (wheel=-2;wheel<=2; wheel++ ){
+   for (sector=0;sector<12; sector++ ){
+     for (station=1; station<=4; station++){
+       bool shifted[7] = {false, false, false, false,false, false, false};
```

Comment:
looks like these should be declared as const and outside the loop

Reply:
actually, these are variables, that change in the loop.